### ISSUE TITLE
No Way to Trace Startup Error

### ISSUE THREAD

**Issue by danjohnso** [aspnet/AspNetCore#14789]
### Is your feature request related to a problem? Please describe.
I am trying to upgrade my project to 3.0 from 2.2.  After following the guide and getting everything to compile, I am hitting a runtime error with no information on how to fix:
```
System.InvalidOperationException: Failed to update route pattern with required values.
   at void Microsoft.AspNetCore.Mvc.Routing.ActionEndpointFactory.AddEndpoints
```

### Describe the solution you'd like
Based on the error message and searching the repo, I narrowed the issue down to this class: 
https://github.com/aspnet/AspNetCore/blob/release/3.0/src/Http/Routing/src/Patterns/DefaultRoutePatternTransformer.cs
Unfortunately, there is not logging in here or in the lead up here:
https://github.com/aspnet/AspNetCore/blob/release/3.0/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

What endpoint is the problem?  I have 100 controllers and 1000+ endpoints but no information on what endpoint it doesn't like.  Some trace logging would be helpful here so I don't have to spend hours commenting out endpoints until it starts working.

**Comment by mkArtakMSFT** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide a minimalistic repro project that illustrates the problem.

**Comment by danjohnso** [comment]
Wasn't sure how to repro this since not being able to isolate the issue is the reason for this request.  I ended up removing all controllers and then adding them back in chunks until I found the broken endpoints.  Narrowed it down to a group of them which had signatures like this:

```
[HttpPost("{id:guid}/action/{action:int}")]
public async Task<IActionResult> Action(Guid id, ProjectActionType action)
```

I have a couple instances where I was using int as a constraint for enums which worked in 2.2, but caused that startup exception in 3.0.  

I assume this is a subtle difference in the new endpoint routing making sure all of the endpoints are route-able ahead of time, which is great, but the process of finding which particular endpoint was the problem was painful across a large project.  There was no indication of what endpoint it was unhappy with from logs.  Controller, method, route template etc, anything would be helpful in tracking this down and would be helpful for both version migrations and even adding new controllers/actions which might conflict with existing routes or have invalid constraints.
